pyqtgraph/graphicsItems/HistogramLUTItem.py

```diff
@@ -25,177 +26,268 @@
-            PlotCurveItem(pen=(255, 0, 0, 100), compositionMode=add),  # r
-            PlotCurveItem(pen=(0, 255, 0, 100), compositionMode=add),  # g
-            PlotCurveItem(pen=(0, 0, 255, 100), compositionMode=add),  # b
-            PlotCurveItem(pen=(200, 200, 200, 100), compositionMode=add),  # a
-            ]
-        
+            PlotCurveItem(pen=(255, 0, 0, 100), compositionMode=comp),  # r
+            PlotCurveItem(pen=(0, 255, 0, 100), compositionMode=comp),  # g
+            PlotCurveItem(pen=(0, 0, 255, 100), compositionMode=comp),  # b
+            PlotCurveItem(pen=(200, 200, 200, 100), compositionMode=comp),  # a
+        ]
         self.plot = self.plots[0]  # for backward compatibility.
         for plot in self.plots:
-            plot.setRotation(90)
+            if self.orientation == 'vertical':
+                plot.setRotation(90)
             self.vb.addItem(plot)
-        
+
         self.fillHistogram(fillHistogram)
         self._showRegions()
-            
-        self.vb.addItem(self.plot)
+
         self.autoHistogramRange()
-        
+
         if image is not None:
             self.setImageItem(image)
-        
+
     def fillHistogram(self, fill=True, level=0.0, color=(100, 100, 200)):
+        """Control fill of the histogram curve(s).
+
+        Parameters
+        ----------
+        fill : bool, optional
+            Set whether or not the histogram should be filled.
+        level : float, optional
+            Set the fill level. See :meth:`PlotCurveItem.setFillLevel
+            <pyqtgraph.PlotCurveItem.setFillLevel>`. Only used if ``fill`` is True.
+        color : color, optional
+            Color to use for the fill when the histogram ``levelMode == "mono"``. See
+            :meth:`PlotCurveItem.setBrush <pyqtgraph.PlotCurveItem.setBrush>`.
+        """
         colors = [color, (255, 0, 0, 50), (0, 255, 0, 50), (0, 0, 255, 50), (255, 255, 255, 50)]
-        for i,plot in enumerate(self.plots):
+        for color, plot in zip(colors, self.plots):
             if fill:
                 plot.setFillLevel(level)
-                plot.setBrush(colors[i])
+                plot.setBrush(color)
             else:
                 plot.setFillLevel(None)
-        
+
     def paint(self, p, *args):
+        # paint the bounding edges of the region item and gradient item with lines
+        # connecting them
         if self.levelMode != 'mono' or not self.region.isVisible():
             return
-        
+
         pen = self.region.lines[0].pen
-        rgn = self.getLevels()
-        p1 = self.vb.mapFromViewToItem(self, Point(self.vb.viewRect().center().x(), rgn[0]))
-        p2 = self.vb.mapFromViewToItem(self, Point(self.vb.viewRect().center().x(), rgn[1]))
+
+        mn, mx = self.getLevels()
+        vbc = self.vb.viewRect().center()
         gradRect = self.gradient.mapRectToParent(self.gradient.gradRect.rect())
+        if self.orientation == 'vertical':
```

Comment:
Not sure if moving this stuff all up here really improves readability compared to a block of if statements in the draw loop. This is a particular spot where it felt like there's got to be a better way